jsonnet/components/prometheus.libsonnet

```diff
@@ -167,6 +167,10 @@ function(params)
           resourceNames: ['nonroot'],
           verbs: ['use'],
         },
+        {
+          nonResourceURLs: ['/federate'],
+          verbs: ['get'],
+        },
```

Comment:
is this needed?

Reply:
Yep, without this change in the role auth is rejected

Reply:
hmm I'm wondering why it is the case?

Reply:
If /federate is not added there then this error is shown:
I0529 12:28:05.416685       1 request.go:1154] Response Body: {"kind":"SubjectAccessReview","apiVersion":"authorization.k8s.io/v1","metadata":{"creationTimestamp":null,"managedFields":[{"manager":"kube-rbac-proxy","operation":"Update","apiVersion":"authorization.k8s.io/v1","time":"2023-05-29T12:28:05Z","fieldsType":"FieldsV1","fieldsV1":{"f:spec":{"f:groups":{},"f:nonResourceAttributes":{".":{},"f:path":{},"f:verb":{}},"f:user":{}}}}]},"spec":{"nonResourceAttributes":{"path":"/federate","verb":"get"},"user":"system:serviceaccount:openshift-monitoring:prometheus-k8s","groups":["system:authenticated"]},"status":{"allowed":false}}

Reply:
`system:serviceaccount:openshift-monitoring:prometheus-k8s` <- it isn't using the proper certificate?

Reply:
It is using this one https://github.com/openshift/cluster-monitoring-operator/pull/1990
but still rejected

Reply:
https://github.com/brancz/kube-rbac-proxy/tree/master/examples/non-resource-url

Reply:
You are right, just the proxy doesn't show the allowed true :S
trying with a curl:
> GET /federate? HTTP/2
> Host: prometheus-k8s.openshift-monitoring.svc:9092
> User-Agent: curl/7.61.1
> Accept: */*